### ISSUE TITLE
Refactor `linkerd inject` to make use of the new endpoint for injecting the proxy

### ISSUE THREAD

**Issue by alpeb** [linkerd/linkerd2#2288]
Depends on #2286

**Comment by klingerf** [comment]
Just a quick clarifying comment here: we should update `linkerd inject` to make use of the new endpoint, but for `linkerd install` we'll still need to use a library to inject the control plane install config, since the control plane won't be available at the time that the install config is generated.

**Comment by alpeb** [comment]
@klingerf thanks for the clarification, that makes sense.

**Comment by alpeb** [comment]
This has been superseded by #2447
